os/fs/smartfs/smartfs_smart.c

```diff
@@ -1960,11 +1961,10 @@ int smartfs_rename(struct inode *mountpt, const char *oldrelpath, const char *ne
 #ifdef CONFIG_SMARTFS_USE_SECTOR_BUFFER
 		if (oldentry.dfirst == newentry.dsector) {
 			/* We will not use any new entry found, we will overwrite the existing entry but with a new name */
-			strncpy(oldentry.name, newentry.name, fs->fs_llformat.namesize);
-			oldentry.prev_parent = oldentry.dsector;
-			ret = smartfs_writeentry(fs, oldentry, type, mode);
-			if (ret != OK) {
-				fdbg("Error writing new entry\n");
+			smartfs_setbuffer(&readwrite, oldentry.dsector, oldentry.doffset + offsetof(struct smartfs_entry_header_s, name), fs->fs_llformat.namesize, (uint8_t *)newentry.name);
```

Comment:
btw, did you checked oldentry.dsector & oldentry.prev_parent?
I am asking this because writeentry is still needed, because it is newly created entry, so utc should be updated, or we can't distinguish latest one if power failed right after below FS_IOCTL....

Reply:
My miss, utc doesn't have to updated. ignore above comment.